### PR TITLE
Implement basic support for Twitter image services

### PR DESCRIPTION
**Pull request by robxu9**
So there are a lot of image services compatible with this.

### CODE REVIEW

File: src/main/java/com/alta189/deskbin/auth/AuthenticationException.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		super(cause);
+		// TODO Auto-generated constructor stub
+	}
+
+	public AuthenticationException(String message, Throwable cause) {
+		super(message, cause);
+		// TODO Auto-generated constructor stub
+	}
+
+	public AuthenticationException(String message, Throwable cause,
```
Comment (stephenwilliams) [inline src/main/java/com/alta189/deskbin/auth/AuthenticationException.java]: No wrapping please :)

File: src/main/java/com/alta189/deskbin/image/TwitterService.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+			if (e.getStatusCode()==401)
+				throw new AuthenticationException("You need to reauthorize your account!");
+			else
+				throw new AuthenticationException("Twitter is down!");
+		}
+		imageService = MediaProvider.TWITTER;
+	}
+
+	@Override
+	public String getName() {
```
Comment (stephenwilliams) [inline src/main/java/com/alta189/deskbin/image/TwitterService.java]: You should return the name ;)

File: src/main/java/com/alta189/deskbin/image/TwitterService.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			InputStream input = IOUtils.toInputStream(IOUtils.toString(baos.toByteArray(),"UTF-8"),"UTF-8");
+			return upload.upload(FILENAME,input);
+		} catch (Exception e){
+			throw new ImageServiceException(e);
+		}
+	}
+
+	@Override
+	public String upload(List<BufferedImage> image)
+			throws ImageServiceException {
```
Comment (stephenwilliams) [inline src/main/java/com/alta189/deskbin/image/TwitterService.java]: wrapping

File: src/main/java/com/alta189/deskbin/image/TwitterService.java (hunk 2 above)
Comment (robxu9) [inline src/main/java/com/alta189/deskbin/image/TwitterService.java]:
```
* Xu_R facedesks (-.-) (_ _) *wham*
```

**Comment by robxu9** [comment]
fixed.

File: src/main/java/com/alta189/deskbin/auth/AuthenticationException.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	 */
+	private static final long serialVersionUID = -2337295399835722053L;
+
+	public AuthenticationException() {
+		// TODO Auto-generated constructor stub
+	}
+
+	public AuthenticationException(String message) {
+		super(message);
+		// TODO Auto-generated constructor stub
```
Comment (stephenwilliams) [inline src/main/java/com/alta189/deskbin/auth/AuthenticationException.java]: Go ahead and clean up the to-do's in here first

File: src/main/java/com/alta189/deskbin/auth/AuthenticationException.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
+package com.alta189.deskbin.auth;
+
+public class AuthenticationException extends Exception {
+
+	/**
```
Comment (stephenwilliams) [inline src/main/java/com/alta189/deskbin/auth/AuthenticationException.java]: This too

**Pull request merged**